application-templates/starter/src/components/channel-details/channel-details-form.js

```diff
@@ -0,0 +1,96 @@
+import PropTypes from 'prop-types';
+import { useFormik } from 'formik';
+import { useIntl } from 'react-intl';
+import LocalizedTextField from '@commercetools-uikit/localized-text-field';
+import TextField from '@commercetools-uikit/text-field';
+import Spacings from '@commercetools-uikit/spacings';
+import SelectField from '@commercetools-uikit/select-field';
+import { CHANNEL_ROLES } from './constants';
+import validate from './validate';
+import messages from './messages';
+
+const getRoleOptions = Object.keys(CHANNEL_ROLES).map((key) => ({
+  label: CHANNEL_ROLES[key],
+  value: CHANNEL_ROLES[key],
+}));
+
+const ChannelDetailsForm = (props) => {
+  const intl = useIntl();
+  const formik = useFormik({
+    initialValues: props.initialValues,
+    onSubmit: props.onSubmit,
+    validate,
+    enableReinitialize: true,
+  });
+
+  const formElements = (
+    <Spacings.Stack scale="l">
+      <TextField
+        name="key"
+        title={intl.formatMessage(messages.channelKeyLabel)}
+        value={formik.values.key}
+        errors={formik.errors.key}
+        touched={formik.touched.key}
+        onChange={formik.handleChange}
+        onBlur={formik.handleBlur}
+        isReadOnly={props.isReadOnly}
+        renderError={(errorKey) => {
+          switch (errorKey) {
+            case 'duplicate':
+              return intl.formatMessage(messages.duplicateKey);
+            default:
+              return null;
+          }
+        }}
+        isRequired
+      />
+      <LocalizedTextField
+        name="name"
+        title={intl.formatMessage(messages.channelNameLabel)}
+        value={formik.values.name}
+        errors={formik.errors.name}
+        touched={Boolean(formik.touched.name)}
+        onChange={formik.handleChange}
+        onBlur={formik.handleBlur}
+        selectedLanguage={props.dataLocale}
+        isReadOnly={props.isReadOnly}
+      />
+      <SelectField
+        name="roles"
+        title={intl.formatMessage(messages.channelRolesLabel)}
+        value={formik.values.roles}
+        errors={formik.errors.roles}
+        touched={formik.touched.roles}
+        onChange={formik.handleChange}
+        onBlur={formik.handleBlur}
+        isMulti
+        options={getRoleOptions}
+        isReadOnly={props.isReadOnly}
+        isRequired
+      />
+    </Spacings.Stack>
+  );
+
+  return props.children({
+    formElements,
+    isDirty: formik.dirty,
```

Comment:
You forgot to expose the `values`

```suggestion
    values: formik.values,
    isDirty: formik.dirty,
```

Reply:
Thanks 🙏